cpp/src/arrow/ipc/adapter.cc

```diff
@@ -149,10 +149,19 @@ class RowBatchWriter {
   }
 
   // This must be called after invoking AssemblePayload
-  int64_t DataHeaderSize() {
-    // TODO(wesm): In case it is needed, compute the upper bound for the size
-    // of the buffer containing the flatbuffer data header.
-    return 0;
+  Status DataHeaderSize(int64_t* size) {
+    // emulates the behavior of Write without actually writing
+    int64_t offset = 0;
```

Comment:
Thanks for your review! (1) is a good solution, probably I'll go with that! Expect to update the PR tonight or some time tomorrow.